ts/sagas/backendInfo.ts

```diff
@@ -50,7 +51,13 @@ function* backendInfoWatcher(): IterableIterator<Effect> {
       // tslint:disable-next-line:saga-yield-return-type
       yield call(startTimer, BACKEND_INFO_LOAD_INTERVAL);
     } else {
-      yield put(backendInfoLoadFailure(new Error("Cannot read server info")));
+      const errorDescription = backendInfoResponse.fold(
+        readableReport,
+        ({ status }) => `response status ${status}`
+      );
```

Comment:
It seems the `readableReport` function return an error if backendInfoResponse isLeft.
(tested switching the device to offline mode). Maybe it could be updated such as:

```````
  while (true) {
    try {
      const backendInfoResponse: SagaCallReturnType<
      typeof getServerInfo
    > = yield call(getServerInfo, {});
      if(backendInfoResponse.isLeft()) {
        throw(backendInfoResponse.value[0].value)
      }

      if (
        backendInfoResponse.isRight() &&
        backendInfoResponse.value.status === 200
      ) {
        yield put(backendInfoLoadSuccess(backendInfoResponse.value.value));
        setInstabugUserAttribute(
          "backendVersion",
          backendInfoResponse.value.value.version
        );
        // tslint:disable-next-line:saga-yield-return-type
        yield call(startTimer, BACKEND_INFO_LOAD_INTERVAL);
      } else {
        const errorDescription = backendInfoResponse.fold(
          readableReport,
          ({ status }) => `response status ${status}`
        )
        throw(new Error(errorDescription));
      }

    } catch(e) {
      yield put(backendInfoLoadFailure(e));
      // tslint:disable-next-line:saga-yield-return-type
      yield call(startTimer, BACKEND_INFO_RETRY_INTERVAL);
    }
  }
}

``````

Reply:
@valentinacala nice catch!
This is a bug about italia-ts-commons https://github.com/teamdigitale/io-ts-commons/commit/39f4768f72c8bb80316126bbdd871e3d3c109498
I updated italia-ts-commons to the last version and the problem seems gone. I guess the code could be just left as is